src/Model/UserItem.php

```diff
@@ -0,0 +1,281 @@
+		if ($thread['ignored']) {
+			return;
+		}
+
+		$notification_type = self::NOTIF_NONE;
+
+		if (self::checkShared($item, $uid)) {
+			$notification_type = $notification_type | self::NOTIF_SHARED;
+		}
+
+		$profiles = self::getProfileForUser($uid);
+
+		// Fetch all contacts for the given profiles
+		$contacts = [];
+		$ret = DBA::select('contact', ['id'], ['uid' => 0, 'nurl' => $profiles]);
+		while ($contact = DBA::fetch($ret)) {
+			$contacts[] = $contact['id'];
+		}
+		DBA::close($ret);
+
+		// Don't create notifications for user's posts
+		if (in_array($item['author-id'], $contacts)) {
+			return;
+		}
+
+		if (self::checkImplicitMention($item, $profiles)) {
+			$notification_type = $notification_type | self::NOTIF_IMPLICIT_TAGGED;
+		}
+
+		if (self::checkExplicitMention($item, $profiles)) {
+			$notification_type = $notification_type | self::NOTIF_EXPLICIT_TAGGED;
+		}
+
+		if (self::checkCommentedThread($item, $contacts)) {
+			$notification_type = $notification_type | self::NOTIF_THREAD_COMMENT;
+		}
+
+		if (self::checkDirectComment($item, $uid, $contacts)) {
+			$notification_type = $notification_type | self::NOTIF_DIRECT_COMMENT;
+		}
+
+		if (self::checkCommentedParticipation($item, $contacts)) {
+			$notification_type = $notification_type | self::NOTIF_COMMENT_PARTICIPATION;
+		}
+
+		if (self::checkActivityParticipation($item, $contacts)) {
+			$notification_type = $notification_type | self::NOTIF_ACTIVITY_PARTICIPATION;
+		}
+
+		if (empty($notification_type)) {
+			return;
+		}
+
+		Logger::info('Set notification', ['iid' => $item['id'], 'uid' => $uid, 'notification-type' => $notification_type]);
+
+		DBA::update('user-item', ['notification-type' => $notification_type], ['iid' => $item['id'], 'uid' => $uid], true);
+	}
+
+	/**
+	 * Fetch all profiles (contact URL) of a given user
+	 * @param int $uid User ID
+	 *
+	 * @return array Profiles
+	 */
+	private static function getProfileForUser($uid)
+	{
+		$notification_data = ['uid' => $uid, 'profiles' => []];
+		Hook::callAll('check_item_notification', $notification_data);
+
+		$profiles = $notification_data['profiles'];
+
+		$user = DBA::selectFirst('user', ['nickname'], ['uid' => $uid]);
+		if (!DBA::isResult($user)) {
+			return [];
+		}
+
+		$owner = DBA::selectFirst('contact', ['url'], ['self' => true, 'uid' => $uid]);
+		if (!DBA::isResult($owner)) {
+			return [];
+		}
+
+		// This is our regular URL format
+		$profiles[] = $owner['url'];
+
+		// Notifications from Diaspora are often with an URL in the Diaspora format
+		$profiles[] = DI::baseUrl().'/u/'.$user['nickname'];
+
+		$profiles2 = [];
+
+		foreach ($profiles AS $profile) {
+			// Check for invalid profile urls. 13 should be the shortest possible profile length:
+			// http://a.bc/d
+			// Additionally check for invalid urls that would return the normalised value "http:"
+			if ((strlen($profile) >= 13) && (Strings::normaliseLink($profile) != 'http:')) {
```

Comment:
I'd rather you remove items from  `$profiles` than have two parallel arrays.

Reply:
This is some copied code from the old notification functionality that I just copied. I kept it this way because it worked. But I can have a look at it.

Reply:
I had a look at the functionality. I slightly improved it by renaming the variables and improving the code standards. But I think that this is the easiest way doing it. We have to add the different formats (SSL and non SSL) and have to prevent duplicates and not to add invalid stuff.

Reply:
Okay, I rewrote that part.